### DISCUSSION TITLE
Unable to start (timeout) in a very old version of tmux 3.0a + kitty

### DISCUSSION THREAD

**Discussion by a-torgovitsky** [sxyazi/yazi#1661] [1. Q&A] [answered]
I recently updated yazi from 0.2.5 to 0.3.3.
With 0.2.5, loading was instantaneous.
With 0.3.3. suddenly there is a 5-10second delay everytime I try to run yazi.
Any ideas what could be going on here?
I am using default configuration.

**Comment by AnirudhG07** [comment]
Can you explain in detail what kind of loading here are you talking about?
I am also using 0.3.3 and i dont face any particular delay issues as such.
Also would be great to send your `yazi --debug` just in case.

**Comment by a-torgovitsky** [comment]
This is simply `yazi` from the command line in an empty directory.
It's very noticeable, and I haven't been able to find anyone else commenting on this, so it's probably idiosyncratic to my system.

Here's what I get with `yazi --debug` (which also has a 5-10 second delay before reporting the following output):
```
yazi --debug                                                                                       ─╯

Yazi
    Version: 0.3.3 (7222e17 2024-09-19)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: ya 0.2.5

Emulator
    Emulator.via_env: ("tmux-256color", "")
    Emulator.via_csi: Ok(Unknown([KittyOld]))
    Emulator.detect : Kitty

Adapter
    Adapter.matches: Kitty

Desktop
    XDG_SESSION_TYPE           : Some("x11")
    WAYLAND_DISPLAY            : None
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/usr/bin/zsh")
    EDITOR             : Some("nvim")
    VISUAL             : Some("nvim")
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : true
    tmux version       : tmux 3.0a
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.2
    magick           : 7.1.1-35
    fzf              : 0.54.2
    fd               : 10.1.0
    rg               : 14.1.0
    chafa            : 1.14.2
    zoxide           : 0.9.4
    7z               : 16.02
    7zz              : 24.07
    jq               : 1.7.1


--------------------------------------------------
When reporting a bug, please also upload the `yazi.log` log file - only upload the most recent content by time.
You can find it in the "/home/at/.local/state/yazi" directory.
```

This is what shows up in the `yazi.log` file:
```
  2024-09-20T12:36:40.781289Z ERROR yazi_adapter::emulator: read_until_da1 timed out: [27, 91, 60, 51, 53, 59, 52, 55, 59, 49, 54, 77, 27, 91, 60, 51, 53, 59, 52, 56, 59, 49, 54, 77, 27, 91, 60, 51, 53, 59, 53, 49, 59, 49, 54, 77, 27, 91, 60, 51, 53, 59, 53, 54, 59, 49, 53, 77, 27, 91, 60, 51, 53, 59, 54, 50, 59, 49, 53, 77, 27, 91, 60, 51, 53, 59, 54, 57, 59, 49, 52, 77, 27, 91, 60, 51, 53, 59, 55, 55, 59, 49, 52, 77, 27, 91, 60, 51, 53, 59, 56, 53, 59, 49, 51, 77, 27, 91, 60, 51, 53, 59, 57, 52, 59, 49, 51, 77, 27, 91, 60, 51, 53, 59, 49, 48, 50, 59, 49, 50, 77], error: Elapsed(())
    at yazi-adapter/src/emulator.rs:213

  2024-09-20T12:38:07.226238Z ERROR yazi_adapter::emulator: read_until_da1 timed out: [27, 95, 71, 105, 61, 51, 49, 59, 79, 75, 27, 92], error: Elapsed(())
    at yazi-adapter/src/emulator.rs:213
```

**Reply by sxyazi** [reply]
It looks like your terminal/multiplexer isn't responding properly to [the DA1 request](https://vt100.net/docs/vt510-rm/DA1.html) Yazi sent. 

I noticed that your `tmux` version is quite outdated; try upgrading it to the latest version 3.4. Also, what terminal are you using? Is it the latest version?

**Reply by a-torgovitsky** [reply]
Thanks for the hint.
I have stopped using `tmux`.
I am using `kitty` and that seems to be the source of the problem.
Loading times are very fast with `alacritty` or GNOME terminal.
I _was_ using a very old version of `kitty`, but I just updated it to 0.36.2 and I still see the problem.
I tried removing all of my `kitty` configs so that all of the settings are default and I still see the problem.

Any other ideas? Thanks again!

**Reply by sxyazi** [reply]
> I was using a very old version of kitty, but I just updated it to 0.36.2 and I still see the problem.

When using the latest version of Kitty, are you still running Yazi inside an older version of Tmux, or outside of Tmux? 

Tmux 3.0a was released 5 years ago and doesn't include some of the newer features that Yazi relies on, which could be causing this issue. 

Also, running `yazi --debug` after upgrading them might help.

**Reply by a-torgovitsky** [reply]
I updated `tmux` to 3.4.
Now if I run `yazi` in `tmux`, the behavior is very fast, as anticipated.

However, I can still reproduce the slow loading behavior if I do this:
1. Start `tmux`.
2. Run `kitty` from `tmux`. A new terminal session is created.
3. Within the new terminal session, run `yazi`.
Here is the output of `yazi --debug` from this approach:

```
╰─❯ yazi --debug                                                                                                     ─╯
[c
Yazi
    Version: 0.3.3 (7222e17 2024-09-19)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: ya 0.2.5

Emulator
    Emulator.via_env: ("xterm-kitty", "tmux")
    Emulator.via_csi: Ok(Unknown([]))
    Emulator.detect : Kitty

Adapter
    Adapter.matches: Kitty

Desktop
    XDG_SESSION_TYPE           : Some("x11")
    WAYLAND_DISPLAY            : None
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/usr/bin/zsh")
    EDITOR             : Some("nvim")
    VISUAL             : Some("nvim")
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : true
    tmux version       : 3.4
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.2
    magick           : 7.1.1-35
    fzf              : 0.54.2
    fd               : 10.1.0
    rg               : 14.1.0
    chafa            : 1.14.2
    zoxide           : 0.9.4
    7z               : 16.02
    7zz              : 24.07
    jq               : 1.7.1


--------------------------------------------------
When reporting a bug, please also upload the `yazi.log` log file - only upload the most recent content by time.
You can find it in the "/home/at/.local/state/yazi" directory.
```
Perhaps I am not understanding something basic, because the debug output still seems to indicate the session is in `tmux`. However, none of the `tmux` features are available in the new `kitty` session.

Meanwhile, in the first window with `tmux`, I see this after trying to load `yazi`:
```
[0.290] Could not move child process into a systemd scope: [Errno 30] Failed to call StartTransientUnit: org.freedesktop.DBus.Error.PropertyReadOnly: Cannot set property OOMPolicy, or unknown property.
[3.081] [PARSE ERROR] Unknown DCS escape code: tmux;␛␛[>q␛␛_Gi=31,s=1,v=1,a=q,t=d,f=24;AAAA␛
[3.081] [PARSE ERROR] Unknown char after ESC: 0x1b
[3.081] [PARSE ERROR] Unknown char after ESC: 0x5c
[65.534] [PARSE ERROR] Unknown DCS escape code: tmux;␛␛[>q␛␛_Gi=31,s=1,v=1,a=q,t=d,f=24;AAAA␛
[65.534] [PARSE ERROR] Unknown char after ESC: 0x1b
[65.534] [PARSE ERROR] Unknown char after ESC: 0x5c
[79.326] [PARSE ERROR] Unknown DCS escape code: tmux;␛␛[?12$p
[79.326] [PARSE ERROR] Unknown DCS escape code: tmux;␛␛P$q q␛
[79.327] [PARSE ERROR] Unknown char after ESC: 0x5c
[79.327] [PARSE ERROR] Unknown DCS escape code: tmux;␛␛[?u␛␛[c
```

Alternatively, if I do this, then I see the expected fast behavior:
```
1. Start `alacritty`.
2. Run `kitty` from `alacritty` session. A new terminal session is created.
3. Within the new terminal session, run `yazi`.
```

So it seems like there is potentially some issue about starting a `kitty` instance from within `tmux` and then trying to run `yazi`.
This is not something I would ever need to do, so for my purposes updating `kitty` and `tmux` solved all of the unexpected behavior. Perhaps this is not something _anyone_ would ever need to do! I just leave it here in case it is useful.

Thank you for the help and your work on this excellent project!

**Reply by sxyazi** [reply, marked as answer]
This is because when launching `kitty` within `tmux`, `kitty`, as a child process of `tmux`, inherits `tmux`'s environment variables (such as `$TMUX` and `TMUX_PANE`), and the older version of Yazi relied on these variables to detect whether the user was in `tmux`.

The issue should now be resolved in the latest version of Yazi, since Yazi no longer depends on these environment variables, but instead uses the CSI sequence to detect whether the user is in `tmux`, which is more reliable.
